src/components/Articles/VoteButton.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from "react"
+import axiosInstance from "../utilities/axios"
+
+const VoteButton = ({ articleId, votes, setVotes }) => {
+  const [loading, setLoading] = useState(false)
+
+  const handleVote = async (increment) => {
+    try {
+      setLoading(true)
+
+      // optimistic
+      setVotes((prevVotes) => prevVotes + (increment ? 1 : -1))
+
+      //   await axiosInstance.patch(`/articles/${articleId}`, { increment })
+
+      setLoading(false)
+    } catch (error) {
+      console.error("Error voting:", error)
+
+      setVotes((prevVotes) => prevVotes - (increment ? 1 : -1))
+
+      alert("Failed to vote. Please try again later.")
+      setLoading(false)
+    }
+  }
+
+  return (
+    <div>
+      <button onClick={() => handleVote(true)} disabled={loading}>
+        Upvote
+      </button>
+      <span>{votes}</span>
+      <button onClick={() => handleVote(false)} disabled={loading}>
```

Comment:
I like that you have one handleVote function which take a value of true or false rather than a separate function for each, has the potential to be a lot DRYer 🙂